src/_pytest/runner.py

```diff
@@ -339,19 +339,16 @@ def from_call(
             function, instead of being wrapped in the CallInfo.
         """
         excinfo = None
-        start = timing.time()
-        precise_start = timing.perf_counter()
+        instant = timing.Instant()
         try:
             result: TResult | None = func()
         except BaseException:
             excinfo = ExceptionInfo.from_current()
             if reraise is not None and isinstance(excinfo.value, reraise):
                 raise
             result = None
-        # use the perf counter
-        precise_stop = timing.perf_counter()
-        duration = precise_stop - precise_start
-        stop = timing.time()
+        duration = instant.elapsed_s()
+        start, stop = instant.interval()
```

Comment:
Unless we make something to handle a native time interval disconnected calls to duration/interval may in theory introduce inaccuracy

Reply:
Additionally we may want to use instants to represent start/stop

We ought to validate if we can mirror the api of preexisting libraries fir convergence

Reply:
> Unless we make something to handle a native time interval disconnected calls to duration/interval may in theory introduce inaccuracy

`interval()` uses `time.time()`, so it is not as precise as `elapsed_s()`... can you elaborate what you mean? 

> Additionally we may want to use instants to represent start/stop

That's what I did in a few places, do you mean I missed some places, or something else? Can you elaborate?

Reply:
what i meant that multiple bits in the same code try to act on the same "interval" but dynamically generate it instead of having a concrete interval and getting its components

Reply:
Got it!